iam-proxy-italia-project/backends/cieoidc/endpoints/authorization_endpoint.py

```diff
@@ -0,0 +1,186 @@
+    _OIDC_AUTHORIZATION_URL = "authorization"
+
+    def __init__(
+            self,
+            config: dict,
+            internal_attributes: dict[str, dict[str, str | list[str]]],
+            base_url: str,
+            name: str,
+            auth_callback_func: Callable[[Context, InternalData], Response],
+            converter: AttributeMapper,
+            trust
+    ) -> None:
+        logger.debug(
+            f"Initializing: {self.__class__.__name__}."
+        )
+        super().__init__(config, internal_attributes, base_url, name, auth_callback_func, converter)
+        self._entity_type = self.config.get("entity_type")
+        self._jwks_core= self.config.get("jwks_core")
+
+
+    @property
+    def _jwks(self) -> dict:
+        _dic_jwks: dict[str, dict] = {self._entity_type: {}}
+        _dic_jwks[self._entity_type]["jwks"] = [public_jwk_from_private_jwk(_k) for _k in self._jwks_core]
+        return _dic_jwks
+
+    def _require_config_field(self, path, label):
+        value = self.config
+        try:
+            for key in path:
+                value = value[key]
+        except (KeyError, TypeError):
+            raise ValueError(f"{label} is missing in {self.__class__.__name__}")
+        if not value:
+            raise ValueError(f"{label} is empty in {self.__class__.__name__}")
+        return value
+
+    def _validate_configs(self):
+        """
+        Validates essential configuration fields for the authorization endpoint.
+        """
+        self._require_config_field(
+            ["endpoints", "authorization_endpoint"], "Authorization endpoint")
+        self._require_config_field(
+            ["endpoints", "authorization_endpoint", "config"], "Authorization endpoint config")
+        self._require_config_field(
+            ["endpoints", "authorization_endpoint", "config", "metadata"], "Metadata")
+        self._require_config_field(
+            ["endpoints", "authorization_endpoint", "config", "metadata", "openid_relying_party"],"OpenId Relying Party")
+        self._require_config_field(
+            ["endpoints", "authorization_endpoint", "config", "metadata", "openid_relying_party","client_id"],"Client ID")
+        self._require_config_field(
+            ["endpoints", "authorization_endpoint", "config", "metadata", "openid_relying_party", "redirect_uris"],
+            "Redirect URI")
+
+
+    def endpoint(self, context, *args):
+        """
+        Handles the authentication response from the OP.
+        :type context: satosa.context.Context
+        :type args: Any
+        :rtype: satosa.response.Response
+
+        :param context: SATOSA context
+        :param args: None
+        :return:
+        """
+        logger.debug(
+            f"Entering method: {inspect.getframeinfo(inspect.currentframe()).function}. Params [context {context}]"
+        )
+
+        authz_endpoint = "http://cie-provider.org:8002/oidc/op/authorization" # TODO Insert this property into config file?
```

Comment:
this constant must not be here

openid federation involved the evaluation of a temporal trust chain, built on top f a discovery process, as defined in openid federation and nationa cie oidc guidelines